reflection/serverreflection.go

```diff
@@ -57,8 +58,10 @@ import (
 
 type serverReflectionServer struct {
 	s *grpc.Server
-	// TODO add more cache if necessary
-	serviceInfo map[string]grpc.ServiceInfo // cache for s.GetServiceInfo()
+
+	mu           sync.RWMutex
```

Comment:
What if we called `getSymbols()` from `Register(...)`? That way we won't need this lock.